src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/layout/index.tsx

```diff
@@ -4,7 +4,10 @@ import { ILayout, ILayoutComponent } from '../../features/form/layout';
 import { GenericComponent } from '../../components/GenericComponent';
 
 export function getLayoutComponentById(id: string, layout: ILayout): ILayoutComponent {
-  const component: ILayoutComponent = layout.find((element) => element.id === id) as ILayoutComponent;
+  const component: ILayoutComponent = layout.find((element) => {
+    const match = id.match(`${element.id}(-[0-9]*)*$`);
```

Comment:
@nkylstad I guess this line might be worthy of a comment in the code? For the next dev stumbling across this code... :)

```
const match = id.match(`${element.id}(-[0-9]*)*$`);
```

Reply:
Added a separate function to do the match, so it can be reused. Also added a description of what this actually checks for.